presto-docs/src/main/sphinx/presto_cpp/properties.rst

```diff
@@ -0,0 +1,157 @@
+===============================
+Presto C++ Properties Reference
+===============================
+
+This section describes Presto C++ configuration properties.
+
+The following is not a complete list of all configuration and
+session properties, and does not include any connector-specific
+catalog configuration properties. For information on catalog 
+configuration properties, see :doc:`Connectors </connector/>`.
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Coordinator Properties
+----------------------
+
+Set the following configuration properties for the Presto coordinator exactly 
+as they are shown in this code block to enable the Presto coordinator's use of 
+Presto C++ workers. 
+
+.. code-block:: none
+
+    native-execution-enabled=true
+    optimizer.optimize-hash-generation=false
+    regex-library=RE2J
+    use-alternative-function-signatures=true
+    experimental.table-writer-merge-operator-enabled=false
+
+These Presto coordinator configuration properties are described here, in 
+alphabetical order. 
+
+``experimental.table-writer-merge-operator-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``boolean``
+* **Default value:** ``true``
+
+  Merge TableWriter output before sending to TableFinishOperator. This property must be set to 
+  ``false`` when used. 
+
+  Optionally, use the session property ``experimental.table-writer-merge-operator-enabled = false``.
+
+``native-execution-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``boolean``
+* **Default value:** ``false``
+
+  This property is required when running Presto C++ workers because of 
+  underlying differences in behavior from Java workers.
+
+  Optionally, use the session property ``native-execution-enabled = true``.
+
+``optimizer.optimize-hash-generation``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``boolean``
+* **Default value:** ``true``
+
+  Set this property to ``false`` when running Presto C++ workers 
+  because Velox does not support optimized hash generation, 
+  instead performing this optimization adaptively.
```

Comment:
I don't think we perform the hash generation optimization adaptively yet.

Reply:
The wording is partially correct since HashTable is adaptive and so the extra hash fields are not needed. Maybe say 
"instead uses a HashTable with adaptive runtime optimizations"

Reply:
I am not clear how the HashTable adaptive runtime optimizations make the extra hash fields redundant.
Can you explain?

Reply:
In java, the hash function was expensive. The extra hash fields were added so that between say a join followed by an agg where the grouping key is the same as the join key, the hash of the keys could be reused.

Since Velox HashTable has adaptive runtime optimizations, the extra hash fields are not used.

Reply:
It is not clear to me if these adaptive runtime optimizations `don't need` or `don't use` the extra hash fields. We should clarify in the doc here.

Reply:
Using the comments here, I have revised to the following: 
 
"Set this property to ``false`` when running Presto C++ workers.  
  Velox does not support optimized hash generation, instead using a HashTable 
  with adaptive runtime optimizations that does not use extra hash fields. "